src/v/kafka/server/handlers/topics/types.cc

```diff
@@ -87,15 +87,27 @@ get_bool_value(const config_map_t& config, std::string_view key) {
     return std::nullopt;
 }
 
-static std::optional<model::shadow_indexing_mode>
+static model::shadow_indexing_mode
 get_shadow_indexing_mode(const config_map_t& config) {
-    std::optional<model::shadow_indexing_mode> mode;
     auto arch_enabled = get_bool_value(config, topic_property_remote_write);
-    if (arch_enabled && *arch_enabled) {
+    auto si_enabled = get_bool_value(config, topic_property_remote_read);
+
+    // If the topic properties are missing, then use the cluster configs.
+    if (!arch_enabled) {
+        arch_enabled
+          = config::shard_local_cfg().cloud_storage_enable_remote_write();
+    }
+
+    if (!si_enabled) {
+        si_enabled
+          = config::shard_local_cfg().cloud_storage_enable_remote_read();
+    }
+
+    model::shadow_indexing_mode mode = model::shadow_indexing_mode::disabled;
+    if (*arch_enabled) {
         mode = model::shadow_indexing_mode::archival;
     }
-    auto si_enabled = get_bool_value(config, topic_property_remote_read);
-    if (si_enabled && *si_enabled) {
+    if (*si_enabled) {
         mode = mode == model::shadow_indexing_mode::archival
                  ? model::shadow_indexing_mode::full
                  : model::shadow_indexing_mode::fetch;
```

Comment:
I know this is not part of this PR but I wonder if we can just have

```
mode |= model::shadow_indexing_mode::fetch;
```

here.  Maybe it is possible now, that we have a concrete value at this point rather than a possible nullopt?

Reply:
I kind of prefer the more verbose approach.